src/Elliptic/Systems/ScalarGaussBonnet/Equations.hpp

```diff
@@ -0,0 +1,172 @@
+}  // namespace sgb
+/// \endcond
+
+namespace sgb {
+
+/*!
+ * \brief Compute the fluxes \f$F^i=\gamma^{ij}\partial_j u(x)\f$ for the scalar
+ * equation in sGB gravity on a spatial metric \f$\gamma_{ij}\f$.
+ */
+void curved_fluxes(gsl::not_null<tnsr::I<DataVector, 3>*> flux_for_field,
+                   const tnsr::II<DataVector, 3>& inv_conformal_metric,
+                   const tnsr::I<DataVector, 3>& shift,
+                   const Scalar<DataVector>& lapse,
+                   const Scalar<DataVector>& conformal_factor,
+                   const tnsr::i<DataVector, 3>& field_gradient);
+
+/*!
+ * \brief Compute the fluxes \f$F^i=\gamma^{ij}\partial_j u(x)\f$ for the scalar
+ * equation in sGB gravity on a spatial metric \f$\gamma_{ij}\f$ on a face
+ * normal.
+ */
+void face_fluxes(gsl::not_null<tnsr::I<DataVector, 3>*> flux_for_field,
+                 const tnsr::II<DataVector, 3>& inv_conformal_metric,
+                 const tnsr::I<DataVector, 3>& shift,
+                 const Scalar<DataVector>& lapse,
+                 const Scalar<DataVector>& conformal_factor,
+                 const tnsr::i<DataVector, 3>& face_normal,
+                 const Scalar<DataVector>& field);
+
+/*!
+ * \brief Add the source terms \f$S=-\Gamma^i_{ij}v^j\f$ for the scalar equation
+ * in sGB gravity arising from the covariant derivative with respect ot the
+ * spatial metric \f$\gamma_{ij}\f$.
+ */
+void add_curved_sources(gsl::not_null<Scalar<DataVector>*> source_for_field,
+                        const tnsr::i<DataVector, 3>& christoffel_contracted,
+                        const tnsr::I<DataVector, 3>& flux_for_field,
+                        const tnsr::i<DataVector, 3>& deriv_lapse,
+                        const Scalar<DataVector>& lapse,
+                        const Scalar<DataVector>& conformal_factor,
+                        const tnsr::i<DataVector, 3>& conformal_factor_deriv);
+
+/*!
+ * \brief Add the sGB coupling term.
+ */
+void add_GB_terms(gsl::not_null<Scalar<DataVector>*> scalar_tensor_equation,
+                  double eps2, double eps4,
+                  const Scalar<DataVector>& weyl_electric,
+                  const Scalar<DataVector>& weyl_magnetic,
+                  const Scalar<DataVector>& field);
+
+/*!
+ * \brief Add sources arising from linearising the sGB coupling term.
+ */
+void add_linearized_GB_terms(
+    gsl::not_null<Scalar<DataVector>*> linearized_scalar_tensor_equation,
+    double eps2, double eps4, const Scalar<DataVector>& weyl_electric,
+    const Scalar<DataVector>& weyl_magnetic, const Scalar<DataVector>& field,
+    const Scalar<DataVector>& field_correction);
+
+/*!
+ * \brief Compute the fluxes \f$F^i\f$ for the scalar equation in sGB gravity on
+ * a spatial metric \f$\gamma_{ij}\f$.
+ */
+struct Fluxes {
+  using argument_tags = tmpl::list<
+      Xcts::Tags::InverseConformalMetric<DataVector, 3, Frame::Inertial>,
+      sgb::Tags::RolledOffShift, gr::Tags::Lapse<DataVector>,
```

Comment:
Can you just use `gr::Tags::Shift` and decide at initialization if you want to apply a rolloff or other things?